src/Model/SearchTrait.php

```diff
@@ -45,22 +44,24 @@ trait SearchTrait
      * Callback fired from the controller.
      *
      * @param \Cake\Datasource\QueryInterface $query Query.
-     * @param array $options The options for the find.
-     *   - `search`: Array of search arguments.
-     *   - `collection`: Filter collection name.
+     * @param array $search Array of search arguments.
+     * @param string $collection Filter collection name.
      * @return \Cake\Datasource\QueryInterface The Query object used in pagination.
      * @throws \Exception When missing search arguments.
      */
-    public function findSearch(QueryInterface $query, array $options): QueryInterface
-    {
-        if (!isset($options['search'])) {
+    public function findSearch(
+        QueryInterface $query,
+        array $search = [],
```

Comment:
Remove the default value. Calling this method without search params would be pointless.